### PR TITLE
🐛 Fix a panic when getExternalRemediationRequest returns nil

### PR DESCRIPTION
**Pull request by goushicui**
:bug:, patch and bugfixes

**What this PR does / why we need it**:
fix bug
obj, err := r.getExternalRemediationRequest(ctx, m, t.Machine.Name) 
when obj is nil will cause panic
obj.GetDeletionTimestamp()

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: controllers/machinehealthcheck_controller.go
Comment (neolit123) [inline controllers/machinehealthcheck_controller.go:304]: you could do something like this, to avoid duplicating the message:

```suggestion
					wrappedErr := errors.Wrapf(err, "failed to fetch remediation request for machine %q in namespace %q within cluster %q", t.Machine.Name, t.Machine.Namespace, t.Machine.ClusterName)
					logger.Error(wrappedErr)
					errList = append(errList, wrappedErr)
```

Comment (goushicui) [reply controllers/machinehealthcheck_controller.go]: updated , thks

**Comment by goushicui** [comment]
@neolit123 updated, thks

**Review by neolit123 — COMMENTED** [review]
/ok-to-test

LGTM, the change makes sense to me.
leaving it to the maintainers for approval.

**Comment by fabriziopandini** [comment]
@goushicui changes lgtm, but it will be great if you can file an issue describing the problem/symptoms you were facing before implementing this fix

**Comment by goushicui** [comment]
@fabriziopandini  I think probably not necessary, it's a minor fix MR when I development and use ERC

File: controllers/machinehealthcheck_controller.go
Comment (vincepri) [inline controllers/machinehealthcheck_controller.go:305]: Do we need to log this error out?

File: controllers/machinehealthcheck_controller.go
Comment (alexander-demichev) [inline controllers/machinehealthcheck_controller.go:303]: I have a question about error wrapping, isn't the recommended way to wrap errors after go 1.13(https://blog.golang.org/go1.13-errors) is using `fmt.Errorf()` ? If so, can we prefer it over `errors.Wrapf()` ?

Comment (vincepri) [reply controllers/machinehealthcheck_controller.go:303]: Usually we don't use stdlib errors, but pkg/error instead

Comment (alexander-demichev) [reply controllers/machinehealthcheck_controller.go:303]: @vincepri is that for historical reasons?

File: controllers/machinehealthcheck_controller.go
Comment (goushicui) [reply controllers/machinehealthcheck_controller.go:305]: need

Comment (vincepri) [reply controllers/machinehealthcheck_controller.go:305]: I don't think we necessarily need to log it out here, given that's going to be logged in the controller already. The best way to surface these errors would be in an condition though.

File: controllers/machinehealthcheck_controller.go
Comment (vincepri) [reply controllers/machinehealthcheck_controller.go:303]: Partially, yes. Most of the k/k repositories use pkg/errors

File: controllers/machinehealthcheck_controller.go
Comment (goushicui) [reply controllers/machinehealthcheck_controller.go]: @vincepri removed

Comment (goushicui) [reply controllers/machinehealthcheck_controller.go]: /vincepri

**Comment by goushicui** [comment]
ping @vincepri @JoelSpeed 
PTAL, thanks!

**Comment by vincepri** [comment]
/retitle 🐛 Fix a panic when getExternalRemediationRequest returns nil

**Review by vincepri — APPROVED** [review]
/milestone v0.4.0

LGTM

**Review by vincepri — APPROVED** [review]
/approve
/lgtm

**Pull request merged**
